skeleton/src/test/java/com/dropbox/kaiken/skeleton/usermanagement/RealUserStoreTest.kt

```diff
@@ -0,0 +1,123 @@
+package com.dropbox.kaiken.skeleton.usermanagement
+
+import app.cash.turbine.test
+import com.dropbox.kaiken.skeleton.core.SkeletonOauth2
+import com.dropbox.kaiken.skeleton.core.SkeletonUser
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class RealUserStoreTest {
+    private val users = MutableStateFlow(emptySet<SkeletonUser>())
+
+    private fun createUserStore(): UserStore = RealUserStore(users)
+
+    @Test
+    fun `GIVEN WHEN THEN`() = runBlockingTest {
+        // GIVEN
+        val userStore = createUserStore()
+
+        // WHEN
+        val events = userStore.getUserEvents()
+
+        // THEN
+        events.test {
+            assertThat(expectMostRecentItem()).isEqualTo(UsersEvent(emptySet()))
+            expectNoEvents()
+        }
+    }
+
+    @Test
+    fun `GIVEN user store WHEN get user events and one user added THEN emit one user`() = runBlockingTest {
+        // GIVEN
+        val userStore = createUserStore()
+
+        // WHEN
+        val events = userStore.getUserEvents()
+        users.emit(setOf(FAKE_USER_ONE))
+
+        // THEN
+        events.test {
+
+            assertThat(expectMostRecentItem()).isEqualTo(
+                UsersEvent(
+                    setOf(FAKE_USER_ONE),
+                    setOf(
+                        FAKE_USER_ONE
+                    ),
+                    emptySet()
+                )
+            )
+            expectNoEvents()
+        }
+    }
+
+    @Test
+    fun `GIVEN user store WHEN get user events and emit and remove user THEN emit one user`() = runBlockingTest {
+        // GIVEN
+        val userStore = createUserStore()
+
+        // WHEN
+        val events = userStore.getUserEvents()
+
+        users.emit(setOf(FAKE_USER_ONE))
+        users.emit(setOf(FAKE_USER_TWO))
+
+        // THEN
+        events.test {
```

Comment:
turbine lets us write little things like this which I think are nice. Lmk if anyone has concerns.